lib/pbench/server/api/resources/pbench_users.py

```diff
@@ -0,0 +1,426 @@
+import jwt
+from flask import request, jsonify, make_response
+from flask_restful import Resource, abort
+from flask_bcrypt import check_password_hash
+from email_validator import validate_email, EmailNotValidError
+from sqlalchemy.exc import SQLAlchemyError, IntegrityError
+from configparser import NoOptionError
+from pbench.server.database.models.users import UserModel
+from pbench.server.database.models.active_tokens import ActiveTokens
+
+from pbench.server.api.auth import Auth
+
+
+class RegisterUser(Resource):
+    """
+    Abstracted pbench API for registering a new user
+    """
+
+    def __init__(self, config, logger):
+        self.server_config = config
+        self.logger = logger
+        self.bcrypt_log_rounds = int(
+            self.server_config.get("pbench-server", "bycrypt_log_rounds")
+        )
+        try:
+            self.token_expire_duration = self.server_config.get(
+                "pbench-server", "token_expiration_duration"
+            )
+        except NoOptionError:
+            self.token_expire_duration = None
+
+    def post(self):
+        """
+        Post request for registering a new user.
+        This requires a JSON data with required user fields
+        {
+            "username": "username",
+            "password": "password",
+            "firstName": first_name,
+            "lastName": "last_name",
+            "email": "user@domain.com"
+        }
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+
+        :return: JSON Payload
+        if we succeed to add a user entry in database, the returned response_object will look like following:
+            response_object = {
+                    "status": "success", # not present if the request fails
+                    "message": "Successfully registered."/"failure message",
+                }
+        To get the auth token user has to perform the login action
+        """
+        # get the post data
+        user_data = request.get_json()
+        if not user_data:
+            self.logger.warning("Invalid json object: {}", request.url)
+            abort(400, message="Invalid json object in request")
+
+        username = user_data.get("username")
```

Comment:
Hey Nikhil! in the current dashboard implementation for Login, we do not have a username field, we have an "email" field as well as a "password" field

Reply:
I'd prefer both; even though most sites allow you to use your email as the username, it's usually not required. Email addresses can change over time (by preference or necessity) but "name" tends to be a more stable entity.